packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -52,8 +56,33 @@ export class TelemetryBuilder {
   }
 }
 
+export class TelemetryProvider {
+  private static instances = new Map<string, TelemetryService>();
+  public static getInstance(extensionName?: string): TelemetryService {
+    // If there is no parameter, it is in the context of core extension
```

Comment:
defaulting it to core extension when extension name is not provided seems odd and leaves room for bugs when external extensions can call this without api name (if I am reading this code correctly)

Reply:
If I make the parameter mandatory, there will be a bunch of changes in core. And I only need to specify the [interface](https://github.com/forcedotcom/salesforcedx-vscode-einstein-gpt/blob/4ea742bd3f9762386609bd168338e6cbc57dfe8d/src/types/TelemetryService.ts#L42) with the parameter in external extensions. What else do you think is necessary?

Reply:
same question as I asked in during demo. is there an validation needed on the extension name? is is possible to create multiple instances for the same extension?

Reply:
@diyer I do not see it is possible to create multiple instances for the same extension. getInstance is only called once in a extension. Could you give me an example?

Reply:
I agree, was thinking more from the perspective of defensive programing.